Sources/LeafKit/LeafLexer.swift

```diff
@@ -132,123 +141,191 @@ struct LeafLexer {
+                return .parameter(.stringLiteral(read))
             case .space:
-                // skip whitespace
-                let read = src.readWhile { $0 == .space }
-                guard let space = read else { fatalError("disallowed by switch") }
-                return .whitespace(length: space.count)
-            case let x where x == .exclamation
-                && src.peek(aheadBy: 1)?.isValidOperator == false:
-                // for when ! stands alone
-                src.pop()
-                return .parameter(.operator(.not))
-            case let x where x.isValidInParameter:
-                let read = src.readWhile { $0.isValidInParameter }
-                guard let name = read else { fatalError("disallowed by switch") }
-                // this parameter is a tag
-                if src.peek() == .leftParenthesis { return .parameter(.tag(name: name)) }
-                
-                // check if expected parameter type
-                if let keyword = Keyword(rawValue: name) { return .parameter(.keyword(keyword)) }
-                else if let op = Operator(rawValue: name) { return .parameter(.operator(op)) }
-                else if let val = Int(name) { return .parameter(.constant(.int(val))) }
-                else if let val = Double(name) { return .parameter(.constant(.double(val))) }
-                
-                // unknown param type.. var
-                return .parameter(.variable(name: name))
-            default:
-                throw LexerError(src: src, lexed: lexed, reason: .invalidParameterToken(next))
+                let read = src.readWhile { $0 == .space } ?? ""
+                return .whitespace(length: read.count + 1)
+            default: break
+        }
+        
+        // Complex Parameter lexing situations - enhanced to allow non-whitespace separated values
+        // Complicated by overlap in acceptable isValidInParameter characters between possible types
+        // Process from most restrictive options to least to help prevent overly aggressive tokens
+        // Possible results, most restrictive to least
+        // * Operator
+        // * Constant(Int)
+        // * Constant(Double)
+        // * Keyword
+        // * Tag
+        // * Variable
+        
+        // if current character isn't valid for any kind of parameter, something's majorly wrong
+        guard current.isValidInParameter else {
+            throw LexerError(.invalidParameterToken(current), src: src, lexed: lexed)
+        }
+
+        // Test for Operator first - this will only handle max two character operators, not ideal
+        // Can't switch on this, MUST happen before trying to read tags
+        if current.isValidOperator {
+            var op = Operator(rawValue: String(current) + String(src.peek()!))
+            if op == nil { op = Operator(rawValue: String(current)) } else { src.pop() }
+            if op != nil { return .parameter(.operator(op!)) }
+        }
```

Comment:
Nominally this isn't really too bad because it's better to not be super aggressive, but if three-character operands are ever included for some reason, would need to be adjusted accordingly. On the upside if someone's annoying enough to write a template like `#if(!!!!!!!!!!!!!this)`, it will adequately stream a bunch of individual `nots`